src/ImageSharp/Formats/Png/PngEncoderCore.cs

```diff
@@ -101,12 +100,12 @@ internal sealed class PngEncoderCore : IImageEncoderInternals, IDisposable
     /// <summary>
     /// The raw data of previous scanline.
     /// </summary>
-    private IMemoryOwner<byte> previousScanline;
+    private IMemoryOwner<byte> previousScanline = null!;
```

Comment:
Shouldn't this field be nullable instead?

Reply:
Most of the usages of these fields are not null. It is possible to be null only when dispose. So I also used null-forgiving here for the sake of code brevity.

Reply:
You can remove the explicit set to null then from disposal. GC will clean up.